core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -336,13 +345,40 @@ class BlockManagerMasterEndpoint(
   private def removeBlockManager(blockManagerId: BlockManagerId): Unit = {
     val info = blockManagerInfo(blockManagerId)
 
+    // Not removing info from the blockManagerInfo map, but only setting the removal timestamp of
+    // the executor in BlockManagerInfo. This info will be removed from blockManagerInfo map by the
+    // blockManagerInfoCleaner once now() - info.executorRemovalTs > executorTimeoutMs.
+    //
+    // We are delaying the removal of BlockManagerInfo to avoid a BlockManager reregistration
+    // while a executor is shutting. This unwanted reregistration causes inconsistent bookkeeping
+    // of executors in Spark.
```

Comment:
Oh..Just some succinct description should be fine. This's too detailed for the comment. I think you can remove the detailed description below.